### ISSUE TITLE
Flutter integration test timeouts with https://dart-review.googlesource.com/c/sdk/+/149613

### ISSUE THREAD

**Issue by gw280** [dart-lang/sdk#42350]
The Flutter Engine roll into Flutter was failing due to timeouts on our integration tests, and this patch was found to be the root cause:

https://dart-review.googlesource.com/c/sdk/+/149613
https://github.com/dart-lang/sdk/commit/b0484ecbde5e09b0addf73bab016ac988800d888

Failures: https://github.com/flutter/flutter/pull/59380/checks?check_run_id=767354642

cc @rmacnak-google

**Comment by zanderso** [comment]
The dart -> engine roller is currently blocked on this.

**Comment by alexmarkov** [comment]
Revert: https://dart-review.googlesource.com/c/sdk/+/151169

**Comment by jonahwilliams** [comment]
We should probably just fix the test right? IIRC this is intended to test the code generation for the l10n tool, which includes the ability to generate deferred for web applications.

**Comment by rmacnak-google** [comment]
Yeah, after some digging I found some generated code includes

```
  if (kIsWeb) {                                                                                                                                                                          
    return loadLibrary().then((dynamic _) => localizationClosure());                                                                                                                     
  } else {                                                                                                                                                                               
    return SynchronousFuture<AppLocalizations>(localizationClosure());                                                                                                                   
  }
```

which is bogus for the non-web case.

**Comment by jonahwilliams** [comment]
Fixed upstream in flutter with https://github.com/flutter/flutter/pull/59539
